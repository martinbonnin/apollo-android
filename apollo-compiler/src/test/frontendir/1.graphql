# species should be always non-nullable and all shapes should merge
query GetAnimal($bla: Boolean!) {
  friend {
    species
    ... on Cat {
      species @include(if: $bla)
    }
  }
}