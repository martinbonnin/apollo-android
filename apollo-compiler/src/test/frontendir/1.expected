package com.example

import kotlin.String

interface GetAnimal {
  interface QueryData {
    val friend: _.AnimalFriend?

    interface AnimalFriend {
      val species: String

      val asCatFriend: _.CatFriend

      interface CatFriend {
        val species: String?
      }
    }
  }
}
